app/code/Tasks/FeedbackManager/Block/FeedbackAdd.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Tasks\FeedbackManager\Block;
+
+use Magento\Customer\Model\SessionFactory;
+
+class FeedbackAdd extends \Magento\Framework\View\Element\Template
+{
+   
+     /** @var SessionFactory */
+     protected $session;
+
+     /** @var */
+     protected $customerData;
+ 
+     /**
+      * Feedback constructor.
+      * @param Template\Context $context
+      * @param SessionFactory $session
+      * @param array $data
+      */
+ 
+     public function __construct(
+        \Magento\Framework\View\Element\Template\Context $context,
+         SessionFactory $session,
```

Comment:
@gayathri-aravind Is there any reason for using factory instead of singleton?

Reply:
Yes, to create a new session instance for each customer and to prefill their data (firstname, lastnmae, email) in the feedback form. And, if I try to use singleton (Magento\Customer\Model\Session) instead of factory, I am facing an error in the object creation because the 'vendor/magento/framework/ObjectManager/Factory/AbstractFactory' is being called in the Session construct() method.

Reference: https://magento.stackexchange.com/questions/227456/difference-between-session-and-sessionfactory-in-magento-2.

And, the session object cannot be a shared.